javaparser-core/src/main/java/com/github/javaparser/printer/ConcreteSyntaxModel.java

```diff
@@ -160,7 +160,7 @@ private static CsmElement typeArguments() {
         concreteSyntaxModelByClass.put(LocalRecordDeclarationStmt.class, sequence(comment(), child(ObservableProperty.RECORD_DECLARATION)));
         concreteSyntaxModelByClass.put(ReturnStmt.class, sequence(comment(), token(GeneratedJavaParserConstants.RETURN), conditional(ObservableProperty.EXPRESSION, IS_PRESENT, sequence(space(), child(ObservableProperty.EXPRESSION))), semicolon()));
         concreteSyntaxModelByClass.put(YieldStmt.class, sequence(comment(), token(YIELD), conditional(ObservableProperty.EXPRESSION, IS_PRESENT, sequence(space(), child(ObservableProperty.EXPRESSION))), semicolon()));
-        concreteSyntaxModelByClass.put(SwitchEntry.class, sequence(comment(), conditional(ObservableProperty.LABELS, IS_NOT_EMPTY, sequence(token(GeneratedJavaParserConstants.CASE), space(), list(ObservableProperty.LABELS), token(GeneratedJavaParserConstants.COLON)), sequence(token(GeneratedJavaParserConstants._DEFAULT), token(GeneratedJavaParserConstants.COLON))), newline(), indent(), list(ObservableProperty.STATEMENTS, newline(), none(), newline()), unindent()));
+        concreteSyntaxModelByClass.put(SwitchEntry.class, sequence(comment(), conditional(ObservableProperty.TYPE, IS_STMT_GROUP_ENTRY, conditional(ObservableProperty.LABELS, IS_NOT_EMPTY, sequence(token(GeneratedJavaParserConstants.CASE), space(), list(ObservableProperty.LABELS), token(GeneratedJavaParserConstants.COLON)), sequence(token(GeneratedJavaParserConstants._DEFAULT), token(GeneratedJavaParserConstants.COLON))), conditional(ObservableProperty.LABELS, IS_NOT_EMPTY, conditional(ObservableProperty.DEFAULT, FLAG, sequence(token(GeneratedJavaParserConstants.CASE), space(), list(ObservableProperty.LABELS), comma(), space(), token(GeneratedJavaParserConstants._DEFAULT), space(), token(GeneratedJavaParserConstants.ARROW)), sequence(token(GeneratedJavaParserConstants.CASE), space(), list(ObservableProperty.LABELS), token(GeneratedJavaParserConstants.ARROW))), sequence(token(GeneratedJavaParserConstants._DEFAULT), space(), token(GeneratedJavaParserConstants.ARROW)))), newline(), indent(), list(ObservableProperty.STATEMENTS, newline(), none(), newline()), unindent()));
```

Comment:
Can you explain in more detail the purpose of this change (why) and the different cases you want to handle (how)? I understand that it's necessary to modify the CSM but it's still another rather obscure part as it has a big impact on the way the LPP handles differences.

Reply:
This line combines 2 changes: 
* support for switch expression syntax (`case X ->`). This is a fix for an existing bug and would be required even without the other changes in this PR
* support for `case null, default ->` added in Java 21 which is new.

The first point is addressed by the first conditional: 
```
conditional(
    ObservableProperty.TYPE,
    IS_STMT_GROUP_ENTRY,
    ..., // If type == STATEMENT_GROUP, we're in the classic "case X: ..." switch statement
    ..., // Otherwise we're in the new "case X -> ..." switch expression
)
```

`IS_STMT_GROUP_ENTRY` is the new condition I added to do this check. From the [SwitchEntry docs](https://javadoc.io/static/com.github.javaparser/javaparser-core/3.25.9/com/github/javaparser/ast/stmt/SwitchEntry.html), we use `SwitchEntry.type` to distinguish between these cases. The type `STATEMENT_GROUP` corresponds to the classic switch statement, where the other 3 types all correspond to the new switch expression. There may be a simpler way to define the concrete syntax for these alternatives, but I don't understand these components well enough to be confident making big changes. Instead, I decided to stick to what already exists as much as possible.

In the classic switch statement case, nothing has changed, so that's just a copy-paste job from what was already there. In the switch expression case, however, there's my second point above to deal with.

The second conditional handles the regular `default` case (with no labels) vs cases with labels:
```
conditional(
    ObservableProperty.LABELS,
    IS_NOT_EMPTY,
    ..., // If labels is not empty, then we're either in the regular "case X, Y -> ..." case, or the new "case null, default ->" - see below for discussion
    ..., // If labels is empty, we're in the regular "default -> ..." case, so handle as before, but with "->" instead of ":"
)
````

The final conditional is then used to handle the `case null, default -> ...` possibility:
```
conditional(
    ObservableProperty.DEFAULT,
    FLAG,
    ..., // If isDefault is true, expect ", default" after labels
    ..., // Otherwise handle as with switch statement, but "->" instead of ":"
)
```

Reply:
The many nested conditionals are quite complex, but the benefit of checking all these cases explicitly is that there's a fairly direct mapping between these alternatives and what's described in the syntax. It might be possible to simplify and flatten this somewhat, but I don't understand the CSM and various supporting constructs well enough to do this confidently and this will need to be updated further for switch pattern support, so simplifying now might be premature.

Reply:
> * support for switch expression syntax (`case X ->`). This is a fix for an existing bug and would be required even without the other changes in this PR

For greater clarity, it would have been preferable to isolate this case in a PR. Maybe for a future fix ;-)

Reply:
Can you add your comments on the CSM in the code for other users to understand.

Reply:
Can't the FLAG condition solve the problem rather than creating a new, very specific condition (IS_STMT_GROUP_ENTRY)?

Reply:
What would be the changes in this case? I can only think of creating a method that would certainly check the nature of the type (for example isStmtGroupEntry() or something similar). If this is the case, it seems to me that this is the best method for managing the condition. Even if you have to add an internal attribute, this seems to me to be more in line with the spirit of what has been initiated.

Reply:
I've updated the PR with this suggestion. I assumed that only fields could be `ObservableProperties`, but then discovered that I can just create a method as you suggested with the `@DerivedProperty` annotation which ensures everything is generated correctly.